internal/controllers/machinedeployment/machinedeployment_sync.go

```diff
@@ -320,7 +320,6 @@ func (r *Reconciler) computeDesiredMachineSet(ctx context.Context, deployment *c
 	desiredMS.Spec.Template.Annotations = cloneStringMap(deployment.Spec.Template.Annotations)
 
 	// Set all other in-place mutable fields.
-	desiredMS.Spec.MinReadySeconds = ptr.Deref(deployment.Spec.MinReadySeconds, 0)
```

Comment:
It is ok to drop this line, but I think it should be replaced by the in place propagation of deployment.Spec.Template.Spec.MinReadySeconds

Also, deployment.Spec.Template.Spec.MinReadySeconds should be excluded from MachineTemplateDeepCopyRolloutFields

Reply:
Same changes for allowing in place mutation of MinReadySeconds should be implemented also in KCP's:
- syncMachines
- computeDesiredMachine

Note: in KCP we decided to not allow to set MinReadySeconds, but I think we should implement changes above no matter of.

Reply:
What is the number should we set to KCP ? Is it 0, right ?

Reply:
I'm not sure why we should make the change to KCP. If there is no user intent because there is no API field for it, we would implement code to explicitly set the field to nil. Why?

Reply:
I agreed
Now that we decided to have nil in the field definition, it is probably cleaner to not set any value in KCP machines

But let's please add a comment both in  syncMachines  and computeDesiredMachine about the fact that we are intentionally not setting this field for KCP because we consider enough to have machine availability driven by readiness of control plane components (no need to add additional delay)